### PR TITLE
WD-17045 - chore(tests): add API tests

### PR DESCRIPTION
**Pull request by huwshimi**
## Changes

Add tests for all API requests.

https://warthogs.atlassian.net/browse/WD-17045

### CODE REVIEW

File: ui/src/util/api.test.ts
[the remark this answers is not in the document]
Comment (BarcoMasile) [reply ui/src/util/api.test.ts:1]: I think he's testing the return values of the `handler` function

**Review by BarcoMasile — APPROVED** [review]

File: ui/src/util/api.test.ts
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/util/api.test.ts:1]: Yes, this handler wraps all other API calls so I'm just testing that it behaves correctly.

**Comment by huwshimi** [comment]
> Would there be value to add tests for the error cases as well (bad status codes, malformed responses)? Right now we only test for success

The API calls are wrapped by the `handleRequest` function which has it's own success/error tests: https://github.com/canonical/identity-platform-admin-ui/pull/483/files#diff-812e151b9aa5459546ef5e5158791307e400edc299f3d9a0845db3bf560c498d

I was trying to avoid duplicating those tests for the API calls as well, but if you think it would be safer to test for errors at the API call level as well I can add those.

**Comment by nsklikas** [comment]
> > Would there be value to add tests for the error cases as well (bad status codes, malformed responses)? Right now we only test for success
> 
> The API calls are wrapped by the `handleRequest` function which has it's own success/error tests: https://github.com/canonical/identity-platform-admin-ui/pull/483/files#diff-812e151b9aa5459546ef5e5158791307e400edc299f3d9a0845db3bf560c498d
> 
> I was trying to avoid duplicating those tests for the API calls as well, but if you think it would be safer to test for errors at the API call level as well I can add those.

Ok that makes sense, thank you for the explanations, I think we can merge it

**Pull request merged**